relayer/relays/parachain/beefy-listener.go

```diff
@@ -735,20 +741,30 @@ func fetchBundleProof(
 	api *gsrpc.SubstrateAPI,
 	digestItem AuxiliaryDigestItem,
 	bundleIndex int,
+	bundle BasicOutboundChannelMessageBundle,
 ) (MerkleProof, error) {
-	var basicChannelBundleProof MerkleProof
 	var proofHex string
+	var rawProof RawMerkleProof
+	var proof MerkleProof
+
 	err := api.Client.Call(&proofHex, "basicOutboundChannel_getMerkleProof", digestItem.AsCommitment.Hash.Hex(), bundleIndex)
 	if err != nil {
-		return basicChannelBundleProof, fmt.Errorf("call rpc basicOutboundChannel_getMerkleProof(%v, %v): %w", digestItem.AsCommitment.Hash.Hex(), bundleIndex, err)
+		return proof, fmt.Errorf("call rpc basicOutboundChannel_getMerkleProof(%v, %v): %w", digestItem.AsCommitment.Hash.Hex(), bundleIndex, err)
+	}
+
+	err = types.DecodeFromHexString(proofHex, &rawProof)
+	log.Debugf("raw proof: %+v", rawProof)
+	if err != nil {
+		return proof, fmt.Errorf("decode merkle proof: %w", err)
 	}
 
-	err = types.DecodeFromHexString(proofHex, &basicChannelBundleProof)
+	proof, err = NewMerkleProof(rawProof, bundle)
+	log.Debugf("proof: %+v", proof)
 	if err != nil {
-		return basicChannelBundleProof, fmt.Errorf("decode merkle proof: %w", err)
+		return proof, fmt.Errorf("decode merkle proof: %w", err)
 	}
 
-	return basicChannelBundleProof, nil
+	return proof, nil
 }
 
 func bundleIndexForAccount(bundles []BasicOutboundChannelMessageBundle, account *[32]byte) int {
```

Comment:
This method doesn't seem to be used anymore and could probably be removed.

Reply:
`fetchOffchainData` also doesn't seem to be used anymore.

Reply:
Removed `bundleIndexForAccount`, but I can't see `fetchOffchainData` used in [the commit where it was added](https://github.com/Snowfork/snowbridge/commit/18c6225b21782170156729d54a35404d876a2c7b). @vgeddes are there any plans to use `fetchOffchainData`?